### PR TITLE
Task_17

### PR DESCRIPTION
**Pull request by totomka**
Hi Stas) Here is my task 17. I will be waiting for your comments. Thank you!

### CODE REVIEW

File: task_18/toma.lobanova/src/test/java/com/hillel/linkedlist/CustomLinkedListTests.java [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+  }
+
+  private void addList() {
+    for (int i = 1; i < 10; i++) {
+      list.add(i);
+    }
+  }
+
+  private StringBuffer iterate(Iterator iterator) {
+    StringBuffer string = new StringBuffer();
```
Comment (parkhomenko) [inline task_18/toma.lobanova/src/test/java/com/hillel/linkedlist/CustomLinkedListTests.java]: StringBuffer is slower than StringBuilder. If you don't work in multithreaded program it is better to use the second one.

File: task_18/toma.lobanova/src/main/java/com/hillel/linkedlist/CustomLinkedList.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    Node current = first;
+    for (int i = 1; i <= index; i++) {
+      current = current.next;
+    }
+    last.next = current;
+  }
+
+  public boolean hasCycle() {
+    Node current = first;
+    for (int i = 1; i < count; i++) {
```
Comment (parkhomenko) [inline task_18/toma.lobanova/src/main/java/com/hillel/linkedlist/CustomLinkedList.java]: Forgot to mention that for this function we assume that the number of elements is unknown. That would be to easy ;)

Comment (totomka) [inline task_18/toma.lobanova/src/main/java/com/hillel/linkedlist/CustomLinkedList.java]: :)I will try to find a better solution till tomorrow.

**Comment by totomka** [comment]
Thank you very much)

**Pull request merged**